core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -666,7 +667,8 @@ class ReplicaManager(val config: KafkaConfig,
                     recordConversionStatsCallback: Map[TopicPartition, RecordConversionStats] => Unit = _ => (),
                     requestLocal: RequestLocal = RequestLocal.NoCaching,
                     transactionalId: String = null,
-                    transactionStatePartition: Option[Int] = None): Unit = {
+                    transactionStatePartition: Option[Int] = None,
+                    actionQueueAdd: (() => Unit) => Unit = actionQueue.add): Unit = {
```

Comment:
how's actionQueueOperation? we don't add() in CoordinatorPartitionWriter right

Reply:
I used a different approach.